mzLib/Proteomics/Protein/Protein.cs

```diff
@@ -784,37 +783,75 @@ private void RemoveMethionineWhenAppropriateFromExistingProduts(InitiatorMethion
         {
             if (initiatorMethionineBehavior == InitiatorMethionineBehavior.Retain || initiatorMethionineBehavior == InitiatorMethionineBehavior.Variable)
             {
                 //when it's variable, we don't have to add anything here, we'll get an additonal proteolysis product later.
-                if(BaseSequence.Length >= minProductBaseSequenceLength)
+                if (BaseSequence.Length >= minProductBaseSequenceLength)
                 {
-                    _proteolysisProducts.Add(new ProteolysisProduct(1, BaseSequence.Length, "intact proteoform"));
+                    _proteolysisProducts.Add(new ProteolysisProduct(1, BaseSequence.Length, "intact proteoform biomarker"));
                 }
-                
             }
             else if (initiatorMethionineBehavior == InitiatorMethionineBehavior.Cleave)
             {
                 if (BaseSequence.Substring(0, 1) == "M")
                 {
                     if (BaseSequence.Length - 1 >= minProductBaseSequenceLength)
                     {
-                        _proteolysisProducts.Add(new ProteolysisProduct(2, BaseSequence.Length, "intact proteoform"));
+                        _proteolysisProducts.Add(new ProteolysisProduct(2, BaseSequence.Length, "intact proteoform biomarker"));
                     }
                 }
                 else
                 {
                     if (BaseSequence.Length >= minProductBaseSequenceLength)
                     {
-                        _proteolysisProducts.Add(new ProteolysisProduct(1, BaseSequence.Length, "intact proteoform"));
+                        _proteolysisProducts.Add(new ProteolysisProduct(1, BaseSequence.Length, "intact proteoform biomarker"));
+                    }
+                }
+            }
+        }
+
+        public void CleaveOnceBetweenProteolysisProducts(int minimumProductLength = 7)
+        {
+            List<int> cleavagePostions = new();
+            List<int> proteolysisProductEndPositions = _proteolysisProducts.Where(p => p.OneBasedEndPosition.HasValue).Select(p => p.OneBasedEndPosition.Value).ToList();
+
+            if (proteolysisProductEndPositions.Count > 0)
+            {
+                foreach (int proteolysisProductEndPosition in proteolysisProductEndPositions)
+                {
+                    if (_proteolysisProducts.Any(p => p.OneBasedBeginPosition == (proteolysisProductEndPosition + 1)))
+                    {
+                        cleavagePostions.Add(proteolysisProductEndPosition);
+                    }
+                }
+            }
+
+            foreach (int position in cleavagePostions)
+            {
+                if (position - 1 >= minimumProductLength)
+                {
+                    string leftType = "N-terminal Portion of Singly Cleaved Protein" + "(" + 1 + "-" + position + ")";
```

Comment:
Use string interpolation? It's faster and this might be done quite a bit because it's in a loop. https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/tokens/interpolated

`string leftType = $"N-terminal Portion of Singly Cleaved Protein(1-{position})"`

Reply:
done